packages/x-date-pickers/src/AdapterDateFnsV3/AdapterDateFnsV3.ts

```diff
@@ -103,31 +103,32 @@ export class AdapterDateFns
     super({ locale: locale ?? enUS, formats, longFormatters });
   }
 
-  public parse = (value: string, format: string) => {
+  // TODO: remove explicit returns in all methods when date-fns-v3 becomes default
+  public parse = (value: string, format: string): Date | null => {
```

Comment:
Not sure if using just `Date` in these returns would work if client dev re-defines it.
Perhaps should be smth like `PickerValidDateLookup['date-fns']` instead? 
Maintainers please advise 🙏

Reply:
> Not sure if using just `Date` in these returns would work if client dev re-defines `PickerValidDateLookup` value.

I'm pretty sure that it shouldn't be a problem. 👌 
The `PickerValidDateLookup` type is not designed for that behavior and would not work on existing adapters as well, because they are statically typing the `TDate` as `Date`. 🙈 🤷 
https://github.com/mui/mui-x/blob/a10624a9c65d9b3301b1c8bd28b43dc1c2a22f20/packages/x-date-pickers/src/AdapterDateFns/AdapterDateFns.ts#L85

Reply:
Gotcha